Cogs.Publishers/CsSchemaPublisher.cs

```diff
@@ -1063,10 +1010,11 @@ private object Translate(JToken prop, Type objectType)
             if (objectType == typeof(DateTimeOffset))
             {
                 string[] values = prop.ToString().Split(new char[] { ' ', '/', ':', '-', '+', 'T', 'Z' });
-                if (values.Length == 7)
+                if (values.Length == 8)
                 {
-                    return new DateTimeOffset(int.Parse(values[2]), int.Parse(values[0]), int.Parse(values[1]),
-                        int.Parse(values[3]), int.Parse(values[4]), int.Parse(values[5]), new TimeSpan());
+                    return new DateTimeOffset(int.Parse(values[0]), int.Parse(values[1]), int.Parse(values[2]),
+                        int.Parse(values[3]), int.Parse(values[4]), int.Parse(values[5]),
+                        new TimeSpan(int.Parse(values[6]), int.Parse(values[7]), 0));
```

Comment:
if splitting by all those characters, how do you know when the hour or minute offset is negative?

Is it the same length ( 8 ) when it is Z vs -04:30?

Try using the TryParseExact with the various formats, and use the bool that is returned to tell you if it is in the right format for the different datetime types we are reading